### PR TITLE
Migrate fieldname and name to new path prop

### LINKED ISSUE #2367: Fix usage of 'label' vs. 'name' with form components
Extract from `formComponentPropsMixin.js`:

```javascript
   // used as field name for validation and as label (if no fallback label is provided)
    name: {
      type: String,
      required: false,
      default: null
    },

    // fallback label; name is used as label if not provided
    label: {
      type: String,
      required: false,
      default: null
    },
```
The idea was that `name` should normally be provided and `label` optionally only to override the displayed text, in case it needs to deviate from `name`.  However, we have several files, where form components are used with `label` property instead of `name` property (or usage of both although not necessary).

Example from `CampSettings.vue`:
```html
<api-text-field
          fieldname="name"
          :label="$tc('entity.camp.fields.name')"
          vee-rules="required"
          :disabled="disabled" />
<api-text-field
          fieldname="title"
          :name="$tc('entity.camp.fields.title')"
          vee-rules="required"
          :disabled="disabled" />
```

**ToDo**
- [ ] check all usages of label and verify if this should changed to name
- [ ] review if `name` property should be mandatory (any use cases where both label & name are not needed?)

### PR DESCRIPTION
**Pull request by manuelmeister**
Closes #2367
Closes #4376

### CODE REVIEW

File: frontend/src/components/form/api/__tests__/ApiColorPicker.spec.js [hunk 1]
```diff
@@ -52,20 +52,18 @@ describe('An ApiColorPicker', () => {
     await user.click(screen.getByLabelText('Speichern'))
 
     // then
-    await waitFor(async () => {
-      const inputField = await screen.findByLabelText(FIELD_LABEL)
-      expect(inputField.value).toBe(COLOR_2)
-      expect(apiMock.getMocks().patch).toBeCalledTimes(1)
-    })
+    const inputField = await screen.findByLabelText(FIELD_LABEL)
```
Comment (BacLuc) [inline frontend/src/components/form/api/__tests__/ApiColorPicker.spec.js:55]: are you sure that you can remove the waitFor?

File: frontend/src/components/form/base/__tests__/__snapshots__/EColorPicker.spec.js.snap [hunk 2]
```diff
@@ -2,11 +2,16 @@
...
 exports[`An EColorPicker > looks like a color picker > pickerclosed 1`] = `
 <div data-app="true">
-     <div class="e-form-container"
-          name="test"
+     <div class="e-form-container e-picker-base"
+          label="test"
           vee-rules="[object Object]">
           <div class="v-menu">
-               <div class="e-form-container">
+               <div buttonarialabeli18nkey="components.form.base.eColorPicker.openPicker"
```
Comment (BacLuc) [inline frontend/src/components/form/base/__tests__/__snapshots__/EColorPicker.spec.js.snap]: why does this change this much?

File: frontend/src/components/form/base/__tests__/__snapshots__/EColorPicker.spec.js.snap [hunk 3]
```diff
@@ -37,10 +47,15 @@ exports[`An EColorPicker > looks like a color picker > pickerclosed 1`] = `
 
 exports[`An EColorPicker > looks like a color picker > pickeropen 1`] = `
 <div data-app="true">
-     <div class="e-form-container"
-          name="test"
+     <div class="e-form-container e-picker-base"
+          label="test"
           vee-rules="[object Object]">
-          <div class="e-form-container"
+          <div buttonarialabeli18nkey="components.form.base.eColorPicker.openPicker"
```
Comment (BacLuc) [inline frontend/src/components/form/base/__tests__/__snapshots__/EColorPicker.spec.js.snap]: why does this change so much?

File: frontend/src/components/program/FormScheduleEntryItem.vue [hunk 4]
```diff
@@ -5,8 +5,7 @@
         <e-date-picker
           v-model="localScheduleEntry.start"
           value-format="YYYY-MM-DDTHH:mm:ssZ"
-          :name="$tc('components.program.formScheduleEntryItem.start')"
-          vee-id="startDate"
+          path="startDate"
```
Comment (BacLuc) [inline frontend/src/components/program/FormScheduleEntryItem.vue:8]: oh oh:
found somewhere here a bug:
image

Comment (BacLuc) [reply frontend/src/components/program/FormScheduleEntryItem.vue:8]: See https://github.com/manuelmeister/ecamp3/pull/6

**Review by BacLuc — APPROVED** [review]
Else it looks good.
Thank you

File: frontend/src/components/form/api/__tests__/ApiColorPicker.spec.js (hunk 1 above)
Comment (manuelmeister) [reply frontend/src/components/form/api/__tests__/ApiColorPicker.spec.js:55]: Yes. It still works and now we don't have the feedback-loop anymore

File: frontend/src/components/form/base/__tests__/__snapshots__/EColorPicker.spec.js.snap (hunk 3 above)
Comment (manuelmeister) [reply frontend/src/components/form/base/__tests__/__snapshots__/EColorPicker.spec.js.snap]: Fixed this with [69b4f47](https://github.com/ecamp/ecamp3/pull/4746/commits/69b4f47195199988c37424ac31714cb6a49fded9)

File: frontend/src/components/form/base/__tests__/__snapshots__/EColorPicker.spec.js.snap (hunk 2 above)
Comment (manuelmeister) [reply frontend/src/components/form/base/__tests__/__snapshots__/EColorPicker.spec.js.snap]: Fixed this with [69b4f47](https://github.com/ecamp/ecamp3/pull/4746/commits/69b4f47195199988c37424ac31714cb6a49fded9)

**Comment by manuelmeister** [comment]
@carlobeltrame can you please have a look at this? I don't want to keep this in sync with devel for too long.

**Review by carlobeltrame — APPROVED** [review]

**Pull request merged**